File: CS/XtraReportApp/NolockInterceptor.cs
using DevExpress.DataAccess.Sql;
using System.Data;

namespace XtraReportApp {
    internal class NolockInterceptor : IDBCommandInterceptor {
        public void CommandCreated(string queryName, IDbCommand command) {
            command.CommandText += " WITH (NOLOCK)";
        }
    }
}

File: VB/XtraReportApp/NolockInterceptor.vb
Imports DevExpress.DataAccess.Sql
Imports System.Data

Namespace XtraReportApp
	Friend Class NolockInterceptor
		Implements IDBCommandInterceptor

		Public Sub CommandCreated(ByVal queryName As String, ByVal command As IDbCommand) Implements IDBCommandInterceptor.CommandCreated
			command.CommandText &= " WITH (NOLOCK)"
		End Sub

	End Class
End Namespace
